releasenotes/notes/deprecate-extensions-d7ab3fca19962e2e.yaml

```diff
@@ -0,0 +1,38 @@
+---
+deprecations:
+  - |
+    Progress towards deprecating the :mod:`qiskit.extensions` module to streamline the structure
+    of Qiskit. The following objects have been moved to :mod:`qiskit.circuit.library`
+    and will be deprecated in :mod:`qiskit.extensions` in a future release,
```

Comment:
It might be good to have a line saying explicitly that the module as a whole is pending deprecation. I know that the meaning is the same as "will be deprecated" and "progress towards deprecating" but it having the keyword could help is someone gets the `PendingDeprecationWarning` and is reading diagonally (this may or may not have been me during this review)

Reply:
Actually I think all deprecations are proper deprecations now -- so we can just write that the whole module is deprecated 👍🏻

Reply:
are they? I see pending here: https://github.com/Qiskit/qiskit/pull/10725/files#diff-bf2e285d2983dfe16101ec2798b449833428a59785a1e4f70da5281dc439dec9R65-R69

Reply:
Perhaps it's also worth to comment on the deprecated ExtensionError?

Reply:
You're right, the module is pending deprecation, as the the new import path from the circuit library is just added in this release. Only some circuit methods are deprecated directly (I mixed something up with #10797 😅).

Reply:
Fixed in 80ace24